python/pyarrow/tests/util.py

```diff
@@ -329,3 +330,22 @@ def get_raise_signal():
         def raise_signal(signum):
             os.kill(os.getpid(), signum)
         return raise_signal
+
+
+@contextlib.contextmanager
+def signal_wakeup_fd(*, warn_on_full_buffer=False):
+    # Use a socket pair, rather a self-pipe, so that select() can be used
+    # on Windows.
+    r, w = socket.socketpair()
+    old_fd = None
+    try:
+        r.setblocking(False)
+        w.setblocking(False)
+        old_fd = signal.set_wakeup_fd(
+            w.fileno(), warn_on_full_buffer=warn_on_full_buffer)
+        yield r
+    finally:
+        if old_fd is not None:
```

Comment:
If I'm reading the docs right, it seems `old_fd` should be `-1` if it wasn't previously set, not `None`.

Reply:
This is for the case where an exception was raised, so `old_fd` didn't get assigned.

Reply:
Ah, got it, makes sense.